tests/torch/fx/test_model_transformer.py

```diff
@@ -124,3 +131,103 @@ def test_output_insertion_transformation(tuple_output, target_point):
     )
+
+
+def count_constants(model) -> int:
+    num_constant_nodes = 0
+    for node in model.graph.nodes:
+        if node.op == "get_attr":
+            num_constant_nodes += 1
+    return num_constant_nodes
+
+
+def count_nodes_with_shared_constants(model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> int:
+    """
+    Gets the number of nodes which use a shared constant.
+    eg:
+          const
+          /   \
+    node1     node2
+
+    returns 2 (node1, and node2)
+    """
+    num_nodes_with_constant_nodes = 0
+    model_graph: torch.fx.Graph = model.graph
+    for node in model_graph.nodes:
+        nncf_node = nncf_graph.get_node_by_name(node.name)
+        num_consumer_nodes = len(nncf_graph.get_next_nodes(nncf_node))
+        if node.op == "get_attr" and num_consumer_nodes > 1:
+            assert nncf_node.is_shared()
+            num_nodes_with_constant_nodes += num_consumer_nodes
+    return num_nodes_with_constant_nodes
+
+
+def check_is_shared_attribute(model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> int:
+    model_graph: torch.fx.Graph = model.graph
+    from collections import Counter
+
+    targets = Counter([node.target for node in model.graph.nodes if node.op == "get_attr"])
+    print(targets)
+    for node in model_graph.nodes:
+        nncf_node = nncf_graph.get_node_by_name(node.name)
+        if node.op == "get_attr" and targets[node.target] > 1:
+            assert nncf_node.is_shared()
+
+
+def test_is_shared_attribute_before_transformation():
+    model = MultiBranchesConnectedModel()
+    ex_inputs = torch.ones((1, 3, 3, 3))
+    captured_model = _capture_model(model, ex_inputs)
+    nncf_graph = NNCFGraphFactory.create(captured_model)
+    check_is_shared_attribute(captured_model, nncf_graph)
+
+
+def test_create_shared_constant_transformation():
+    model = MultiBranchesConnectedModel()
+    ex_inputs = torch.ones((1, 3, 3, 3))
+    captured_model = _capture_model(model, ex_inputs)
+    assert count_constants(captured_model) == 9
+    shared_constants_unification_transformation(captured_model)
+    nncf_graph = NNCFGraphFactory.create(captured_model)
+    assert count_nodes_with_shared_constants(captured_model, nncf_graph) == 3
+    assert count_constants(captured_model) == 7
+
+
+def get_shared_constant_nodes(nncf_graph: NNCFGraph):
+    """
+    Gets a dict of constant nodes as key and consumer nodes as values which are shared in the model.
+    eg:
+          const
+          /   \
+    node1     node2
+
+    returns ({const:[node1, node2]}) 
+    """
+    shared_const_node_consumer_node = {}
+    for node in nncf_graph.get_all_nodes():
+        consumer_nodes = nncf_graph.get_next_nodes(node)
+        if node.metatype in CONST_NOOP_METATYPES and len(consumer_nodes) > 1:
+            shared_const_node_consumer_node[node] = consumer_nodes
+    return shared_const_node_consumer_node
+
+
+def test_update_shared_constant():
```

Comment:
```suggestion
@pytest.mark.parametrize("unification", [False, True])
def test_update_shared_constant(unification):
```

Reply:
I had created another function for the same, I have added pytest.mark.parametrize now and merged it into a single `test_is_shared_attribute_before_transformation` now.

For this `test_update_shared_constant` function, it simply updates the constant from a consumer node and checks the constants value from the sister node of the said consumer node.